mock/SudokuSolver-SecondTry.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+
+namespace Interviews 
+{
+// time taken: 00:49:06
+
+public class SudokuSolver
+{
+	private const int Size = 9;
+	private const int StartIndex = 0;
+	private const int EndIndex = Size - 1;
+	private const int EmptyValue = 0;
+	private const int MinValue =1;
+	private const int MaxValue =9;
+
+	// returns true if at least one solution for a partially filled sudoku is found
+	public bool HasSolution(byte[,] sudoku)
+	{
+		if (sudoku is null) throw new ArgumentNullException(nameof(sudoku));
+		Validate(sudoku);
+		
+		return HasSolution(sudoku, StartIndex ,StartIndex);			
+
+	}
+
+	private bool HasSolution(byte[,] sudoku, int row, int column)
+	{
+		// handle indexes - TODO: extract a separate index struct / class
+		if (row>EndIndex)
+		{
+			row = StartIndex;
+			column = column + 1;
+		}
+
+		// when to stop - sudoku is filled entirely
+		if (column == EndIndex+1)
+			return true;
+		
+		// respect for filled in values: assume empty - means zero
+		if (sudoku[row, column] == EmptyValue) 
```

Comment:
Error in code - should be inequality
Just like comment says, we are after **filled** in values, not empty